validator/client/src/main/java/tech/pegasys/teku/validator/client/ValidatorTimingActions.java

```diff
@@ -35,6 +40,9 @@ public void onSlot(final UInt64 slot) {
     validatorIndexProvider.lookupValidators();
     blockDuties.onSlot(slot);
     attestationDuties.onSlot(slot);
+    if (slot.mod(SLOTS_PER_EPOCH).equals(UInt64.ONE)) {
```

Comment:
In terms of timing, would it be better to run 1 slot after the epoch transition, so we're more likely to pull in the latest updates from the epoch transition?  I'm thinking there might be some lag in processing the latest block and updating the chain head.

Reply:
That was my exact thinking and why I'm running this method every 
`slot.mod(SLOTS_PER_EPOCH).equals(UInt64.ONE))`
and not 
`slot.mod(SLOTS_PER_EPOCH).equals(UInt64.ZERO))`

Reply:
Am I missing something?

Reply:
ha!  no - I'm just so used to reading `x % y == 0` 🤦‍♀️

Reply:
hahaha